<launch>
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="odom"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan_filtered" />

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" clear_params="true">
    <rosparam file="$(find piko_nav)/config/dwa/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find piko_nav)/config/dwa/costmap_common_params.yaml" command="load" ns="local_costmap" />   
    <rosparam file="$(find piko_nav)/config/dwa/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find piko_nav)/config/dwa/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find piko_nav)/config/dwa/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find piko_nav)/config/dwa/move_base_params.yaml" command="load" />
    <rosparam file="$(find piko_nav)/config/dwa/global_planner_params.yaml" command="load" />
    <rosparam file="$(find piko_nav)/config/dwa/navfn_global_planner_params.yaml" command="load" />
    
    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="cmd_vel" to="navi_velocity_smoother/move_base_raw_cmd_vel"/>

    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>
</launch>
